rpm-package make target is broken in v0.55.0 release

**Describe the bug**
rpm-package make target is broken in v0.55.0 release.

**Steps to reproduce**
```make rpm-package```

**What did you expect to see?**
A generated RPM pacakge.

**What did you see instead?**

```
[mike@fedora opentelemetry-collector-contrib ((tracegen/v0.55.0))]$ make rpm-package
go: downloading go.opentelemetry.io/collector v0.55.0
go: downloading github.com/aws/aws-sdk-go v1.44.47
... (lots more download of go dependencies)....
make otelcontribcol-linux_amd64
make[1]: Entering directory '/home/mike/dev/opentelemetry-collector-contrib'
make[1]: *** No rule to make target 'otelcontribcol-linux_amd64'.  Stop.
make[1]: Leaving directory '/home/mike/dev/opentelemetry-collector-contrib'
make: *** [Makefile:334: rpm-package] Error 2
```

**What version did you use?**
v0.55.0

**What config did you use?**
Config: (e.g. the yaml config file)
default.

**Environment**
Fedora 36

**Additional context**
n/a

--- a/Makefile
+++ b/Makefile
@@ -331,7 +331,7 @@ build-examples:
 .PHONY: deb-rpm-package
 %-package: ARCH ?= amd64
 %-package:
-	$(MAKE) otelcontribcol-linux_$(ARCH)
+	GOOS=linux GOARCH=$(ARCH) $(MAKE) otelcontribcol
 	docker build -t otelcontribcol-fpm internal/buildscripts/packaging/fpm
 	docker run --rm -v $(CURDIR):/repo -e PACKAGE=$* -e VERSION=$(VERSION) -e ARCH=$(ARCH) otelcontribcol-fpm
 

--- a/unreleased/fix-makefile-bug.yaml
+++ b/unreleased/fix-makefile-bug.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: tooling
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Fix a bug in the makefile causing `make rpm-package` to fail"
+
+# One or more tracking issues related to the change
+issues: [ 12162 ]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

